xl2times/datatypes.py

```diff
@@ -70,6 +71,63 @@ def has_tag(cls, tag):
         return tag in cls._value2member_map_
 
 
+class DataModule(str, Enum):
+    """Categorise data into modules based on the file they are coming from."""
+
+    base = "VT_*.*, BY_Trans.*"
+    syssettings = "SysSettings.*"
+    subres = "SubRES_TMPL/SubRES_*.*"
+    sets = "Set*.*"
+    lma = "LMA*.*"
+    demand = "SuppXLS/Demands/Dem_Alloc+Series.*, SuppXLS/Demands/ScenDem_*.*"
```

Comment:
Btw, I also realized that this isn't how Enums are "supposed" to be used. The "value" of an Enum should be a unique value, and the "names" are symbolic names that make the code cleaner/more robust. I think the canonical way of doing this would be to define an enum like 
```python
from enum import Enum, auto
class DataModule(Enum):
    base = auto()
    syssettings = auto()
    ...
```
and then having a mapping from Enum values to regexs/patterns `path_to_DataModule` or something, which is then used in `_find_type` to match the given path and then it returns a `DataModule`. This way, the method `module_type` will not have to return `module_type.name` but can return `module_type` instead, which is a `DataModule` object, which is useful down the line if some other transform wants to use some Enum method/cookbook. Or in order to have more type information in the DataFrames, as the `data_module_type` column can be of type `DataModule` instead of string.

I'm not sure how we're going to be using this though, so feel free to decide what makes most sense for this application and go with that, even if it is the current approach! :)

See also: https://docs.python.org/3/howto/enum.html#enum-cookbook

Reply:
Super, thanks for looking into it! Let's merge as is to keep the pace. When you have the time, could you look into refactoring though?